### PR TITLE
add hash computation

### PR DESCRIPTION
**Pull request by massibartoli**

### CODE REVIEW

File: src/DPU/ecdsa_dpu_hash.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "libsig.h"
+#include <stdio.h>
+#include <string.h>
+
+#include <mram.h>
+#include "ecdsa.h"
+
+__mram_noinit mram_t mram;
+extern __mram_ptr void *__sys_sec_mram_start;
```
Comment (dlagneux-upmem) [inline src/DPU/ecdsa_dpu_hash.c]: do you need it?

File: src/DPU/ecdsa_host.c [hunk 2]
```diff
@@ -91,16 +97,77 @@ int main(void)
         prepare_data(area2);
     }
 
-    // Copy DPU program to MRAM
-    fdbin = open(DPU_BINARY,O_RDONLY);
+    // Copy DPU HASH elf to MRAM
```
Comment (dlagneux-upmem) [inline src/DPU/ecdsa_host.c]: maybe the code can be factorized a little bit

File: src/hash/sha256.c [hunk 3]
```diff
@@ -103,21 +109,36 @@ void sha256_update(sha256_context *ctx, const u8 *input, u32 ilen)
 
 	if ((left > 0) && (remain_ilen >= fill)) {
 		/* Copy data at the end of the buffer */
-		local_memcpy(ctx->sha256_buffer + left, data_ptr, fill);
+		if (((uint32_t)data_ptr >= MRAM_START) && ((uint32_t)data_ptr <=  MRAM_END)) {
+			mram_read((__mram_ptr void *)(uintptr_t)data_ptr, tmp, (fill + (8-(fill%8))));
+			local_memcpy(ctx->sha256_buffer + left, tmp, fill);
```
Comment (dlagneux-upmem) [inline src/hash/sha256.c:114]: Just wondering, can you use our API `__memcpy_wm(tmp, (__mram_ptr void *)(uintptr_t)data_ptr, fill)`?

File: src/DPU/ecdsa_host.c (hunk 2 above)
Comment (massibartoli) [reply src/DPU/ecdsa_host.c]: Done, could you please have a look?

File: src/hash/sha256.c (hunk 3 above)
Comment (massibartoli) [reply src/hash/sha256.c:114]: yes you are right, I will have a try and I'll let you know

**Review by dlagneux-upmem — APPROVED** [review]

**Pull request merged**